gtsfm/runner/gtsfm_runner_base.py

```diff
@@ -202,10 +221,30 @@ def run(self) -> None:
         """Run the SceneOptimizer."""
         start_time = time.time()
 
-        # create dask client
-        cluster = LocalCluster(
-            n_workers=self.parsed_args.num_workers, threads_per_worker=self.parsed_args.threads_per_worker
-        )
+        # create dask cluster
+        if self.parsed_args.run_cluster:
+            workers = OmegaConf.load(
+                os.path.join(
+                    self.parsed_args.output_root, "gtsfm", "configs", self.parsed_args.cluster_config_name))["workers"]
+            scheduler = workers[0]
+            cluster = SSHCluster(
+                [scheduler] + workers,
+                scheduler_options={"dashboard_address": self.parsed_args.dashboard_port},
+                worker_options={
+                    "n_workers": self.parsed_args.num_workers,
+                    "nthreads": self.parsed_args.threads_per_worker,
+                },
+            )
+            client = Client(cluster)
+            # getting first worker's IP address and port to do IO
+            io_worker = list(client.scheduler_info()["workers"].keys())[0]
+            self.loader._input_worker = io_worker
+            self.scene_optimizer._output_worker = io_worker
```

Comment:
I will propose a new design where we just scatter the computation for detector/descriptor/matcher/verifier. You would not need to worry about pinning IO to workers in that case as the most of the stuff will happen outside dask.

Reply:
Scattering the computations would be great! I think it will also improve the runtime. I think we might still need IO workers as the data could be on one machine only and we need a way to restrict data reading/output tasks to the machine that has access to the data.